internal/transport/http2_client.go

```diff
@@ -1215,12 +1225,18 @@ func (t *http2Client) setGoAwayReason(f *http2.GoAwayFrame) {
 			t.goAwayReason = GoAwayTooManyPings
 		}
 	}
+	var m bytes.Buffer
+	m.WriteString("code: ")
+	m.WriteString(f.ErrCode.String())
+	m.WriteString(", debug data: ")
+	m.Write(f.DebugData())
+	t.goAwayDebugMessage = m.String()
```

Comment:
Is this better than `t.goAwayDebugMessage = fmt.Sprintf("code: %s, debug data: %v", f.ErrCode, f.DebugData)` ?

Reply:
That looks better, changed to that